aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/SyncProcessorTest.java

```diff
@@ -116,6 +130,88 @@ public void setup() throws AmplifyException {
             .build();
     }
 
+    /**
+     * During a base sync, there are a series of events that should be emitted.
+     * This test verifies that these events are published via Amplify Hub depending
+     * on actions takes for each available model.
+     * @throws DataStoreException Not expected.
+     */
+    @Test
+    public void dataStoreHubEventsTriggered() throws DataStoreException {
+        // Arrange - BEGIN
+        int expectedModelCount = Arrays.asList(Post.class, BlogOwner.class).size();
+        // Collects one syncQueriesStarted event.
+        HubAccumulator syncStartAccumulator =
+            createAccumulator(syncQueryStartedForModels(modelCount), 1);
+        // Collects one syncQueriesReady event.
+        HubAccumulator syncQueryReadyAccumulator =
+            createAccumulator(forEvent(DataStoreChannelEventName.SYNC_QUERIES_READY), 1);
+        // Collects one modelSynced event for each model.
+        HubAccumulator modelSyncedAccumulator =
+            createAccumulator(forEvent(DataStoreChannelEventName.MODEL_SYNCED), expectedModelCount);
+
+        // Add a couple of seed records so they can be deleted/updated.
+        storageAdapter.save(DRUM_POST.getModel());
+        storageAdapter.save(BLOGGER_ISLA.getModel());
+
+        // Mock sync query results for a couple of models.
+        AppSyncMocking.sync(appSync)
+            .mockSuccessResponse(Post.class, DELETED_DRUM_POST)
+            .mockSuccessResponse(BlogOwner.class, BLOGGER_ISLA, BLOGGER_JAMESON);
+
+        // Start the accumulators.
+        syncQueryReadyAccumulator.start();
+        syncStartAccumulator.start();
+        modelSyncedAccumulator.start();
+
+        TestObserver<ModelWithMetadata<? extends Model>> hydrationObserver = TestObserver.create();
+        // Arrange - END
+
+        // Act: kickoff sync.
+        syncProcessor.hydrate().subscribe(hydrationObserver);
+
+        // Check - BEGIN
+        // Verify that sync completes.
+        assertTrue(hydrationObserver.awaitTerminalEvent(OP_TIMEOUT_MS, TimeUnit.MILLISECONDS));
+        hydrationObserver.assertNoErrors();
+        hydrationObserver.assertComplete();
+
+        // Verify that syncQueriesStarted was emitted once.
+        assertEquals(1, syncStartAccumulator.await((int) OP_TIMEOUT_MS, TimeUnit.MILLISECONDS).size());
```

Comment:
Ah crumbs. I guess `HubAccumulator` should have taken `long`, in hindsight, huh?

Reply:
Yeah...we can always change that. I'll make a note to do that in a follow up PR

Reply:
Sounds good, yes -- for a separate PR!